addons/hr_holidays/tests/test_company_leave.py

```diff
@@ -315,10 +322,10 @@ def test_leave_whole_company_07(self):
         })
         company_leave._compute_date_from_to()
 
-        with self.assertQueryCount(__system__=773, admin=865):
+        with self.assertQueryCount(__system__=797, admin=865):
```

Comment:
Do you know why it's introducing new queries?
Maybe we can check to reduce their number

Reply:
I suspect the fact that the conflicting holidays are retrieved with a search instead of a search_count and after that, the one2many relation with the employees where several information are retrieved from: ids, name,...
Not sure how the number of queries could be reduced except by decreasing the amount of information we retrieve in regards to the conflicting holidays...?

Reply:
It needs to be checked functionally, but something like this should work without losing any details:
https://gist.github.com/kba-odoo/f98370d2d61b661967411efc1c0b6ade

It retrieves all the (potential) conflicting leaves in a single query.
The execution time of `_check_date` goes from 15% to 0.8% of the total time it takes to validate all the leaves :-)

Reply:
Legend... Thank you very much :-) The query count is now back to 773